File: C#/Container With Most Water.cs
public class Solution11 {
    public int MaxArea(int[] height) {
        int S = 0, i = 0, j = height.Length -1;
        while (i < j) {
            S = Math.Max(S, (j - i) * Math.Min(height[i], height[j]));
            if (height[i] < height[j])
            {
                i++;
            }
            else
            {
                j--;
            }
        }
        return S;
    }
}

File: Golang/ContainerWithMostWater.go
func maxArea(height []int) int {
    
    l, r, maxArea := 0, len(height) - 1, 0

    for l < r {
        width := r - l
        maxArea = max(maxArea, min(height[l], height[r]) * width)

        if height[l] <= height[r] {
            l = l + 1
        } else {
            r = r - 1
        }
    }
    
    return maxArea
}

func max(a int, b int) int {
    if a < b {
        return b
    }
    return a
}

func min(a int, b int) int {
    if a < b {
        return a
    }
    return b
}